modules/core/arrow-data/src/main/kotlin/arrow/data/ListK.kt

```diff
@@ -33,6 +33,13 @@ data class ListK<out A>(val list: List<A>) : ListKOf<A>, List<A> by list {
     return Eval.defer { loop(this) }
   }
 
+  override fun equals(other: Any?): Boolean =
+    when (other) {
```

Comment:
I don't disagree, but we should also do it for the other wrappers, and to be honest if we're using 1.3 we should just use inline classes instead.